hadoop-ozone/dev-support/checks/license.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script checks if all third-party dependencies have licenses we can use.
+# Optionally accepts the aggregated third-party license list file to be checked.
+# Otherwise it requires Ozone to be available from Maven repo (can be local),
+# so that it can generate the license list.
+#
+# When adding a new dependency to Ozone with a license that fails to match:
+# * verify that the license is allowed, ref: https://www.apache.org/legal/resolved.html
+# * tweak the patterns to allow
+
+set -euo pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$DIR/../../.." || exit 1
+
+REPORT_DIR=${OUTPUT_DIR:-"$DIR/../../../target/license"}
+mkdir -p "$REPORT_DIR"
+REPORT_FILE="${REPORT_DIR}/summary.txt"
+
+DEFAULT_SRC="target/generated-sources/license/THIRD-PARTY.txt"
+src="${1:-${DEFAULT_SRC}}"
+
+if [[ ! -e ${src} ]]; then
+  MAVEN_OPTIONS="-B -fae -Dskip.npx -Dskip.installnpx --no-transfer-progress ${MAVEN_OPTIONS:-}"
+  mvn ${MAVEN_OPTIONS} license:aggregate-add-third-party | tee "${REPORT_DIR}/output.log"
+  src="${DEFAULT_SRC}"
+fi
+
+L='Licen[cs]e' # sometimes misspelled
+
+# filter all allowed licenses; any remaining item indicates a possible problem
+grep '(' ${src} \
+  | ( grep -i -v \
+    -e "Apache ${L}" -e "Apache Software ${L}" -e "Apache v2" -e "Apache.2" \
+    -e "Bouncy Castle ${L}" \
+    -e "(BSD)" -e "(The BSD ${L})" -e "(BSD.[23]" -e "\<BSD ${L} [23]" -e "\<[23]\>.Clause.\<BSD\>" \
+    -e "(CDDL\>" -e ' CDDL '\
+    -e "(EDL\>" -e "Eclipse Distribution ${L}" \
+    -e "(EPL\>" -e "Eclipse Public ${L}" \
+    -e "\<Go ${L}" \
+    -e "(MIT)" -e "\<MIT ${L}" \
+    -e "Modified BSD\>" \
+    -e "New BSD ${L}" \
+    -e "Public Domain" \
+    -e "Revised BSD\>" \
+    -e "Unknown ${L}" \
```

Comment:
Currently the following dependencies are affected:

```
     (Unknown license) jsp-api (javax.servlet.jsp:jsp-api:2.1 - no url defined)
     (Unknown license) Jettison (org.codehaus.jettison:jettison:1.1 - no url defined)
```

* `jsp-api:2.1` has CDDL+GPL2 dual license ([link](https://github.com/javaee/javaee-jsp-api/blob/master/LICENSE))
* `jettison` has Apache license ([link](https://github.com/codehaus/jettison/blob/master/src/main/resources/META-INF/LICENSE))

So these are allowed (if I found the right sources).  I don't know why the plugin cannot find that information.

The plugin may fail to find the license for some future dependency with forbidden license.  If we want to avoid that, we should allow `jsp-api:2.1` and `jettison:1.1` explicitly, instead of `Unknown License`.

Reply:
Let's have an allowed file to handle these cases?
```
javax.servlet.jsp:jsp-api:2.1 CDDL 1.1
org.codehaus.jettison:jettison:1.1 Apache License 2.0
```
When a dependency is found in this file, it is allowed.

Reply:
Using a separate file is a good idea, so we can record the actual license, too.